### PR TITLE
Check iterator.hasNext() to avoid NoSuchElementEx

### PR DESCRIPTION
**Pull request by sancar**
fixes https://github.com/hazelcast/hazelcast/issues/16031

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientClusterServiceImpl.java [hunk 1]
```diff
@@ -150,8 +151,11 @@ public long getClusterTime() {
...
         final ClientConnectionManager cm = client.getConnectionManager();
-        final ClientConnection connection = cm.getActiveConnections().iterator().next();
-        InetSocketAddress inetSocketAddress = connection != null ? connection.getLocalSocketAddress() : null;
+        InetSocketAddress inetSocketAddress = null;
+        Iterator<ClientConnection> iterator = cm.getActiveConnections().iterator();
+        if (iterator.hasNext()) {
+            inetSocketAddress = iterator.next().getLocalSocketAddress();
+        }
         UUID clientUuid = cm.getClientUuid();
         return new ClientImpl(clientUuid, inetSocketAddress, client.getName(), labels);
```
Comment (ihsandemir) [inline hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientClusterServiceImpl.java]: composing the `ClientImpl ` object with a null address, is this not a problem?

Comment (sancar) [reply hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientClusterServiceImpl.java]: Well, We don't have another chance. And this pr is not changing the behaviour. We were passing null for the address when connection is null anyway.

**Comment by sancar** [comment]
run-lab-run

**Comment by sancar** [comment]
run-lab-run

**Pull request merged**